book/src/getting-started/2-our-first-cxx-qt-module.md

```diff
@@ -0,0 +1,85 @@
+<!--
+SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+SPDX-FileContributor: Leon Matthes <leon.matthes@kdab.com>
+
+SPDX-License-Identifier: MIT OR Apache-2.0
+-->
+
+# Our first CXX-Qt module
+
+As with all things Rust, we'll first want to create a cargo project.
+```bash
+cargo new --lib cxx-qt-getting-started
+```
+
+As outlined in the previous section, to define a new QObject subclass, we'll need to create a Rust module.
+So let's go into the `src/lib.rs` file.
+We'll modify this file until it looks like this:
+
+```rust,noplayground
+{{#include ../../../examples/qml_minimal/src/lib.rs:book_cxx_qt_module}}
+```
+
+This is a lot to take in, so let's go one step at a time.
+Starting with the module definition:
+```rust,noplayground
+{{#include ../../../examples/qml_minimal/src/lib.rs:book_make_qobject_macro}}
+```
+
+Because we add the `#[make_qobject]` macro to the module definition, CXX-Qt will create a new QObject subclass from this module.
+The new QObject subclass in our case will be named `MyObject`, as CXX-Qt automatically converts Rusts snake_case to the Qt default PascalCase.
+CXX-Qt is all about idiomatic code in both Rust and C++, so it will do its best to keep styling consistent for C++ and Rust as well.
+
+For the `#[make_qobject]` macro to work, we first need to define the data that will live in the new C++ object.
+This is done with the `Data` struct:
+```rust,noplayground
+{{#include ../../../examples/qml_minimal/src/lib.rs:book_data_struct}}
+```
+That means the newly created QObject subclass will have two properties as members: `number` and `string`. For names that contain multiple words, like `my_number`, CXX-Qt will again perform the snake_case to camelCase conversion to fit with C++/QML naming conventions.
+
+Note that the data types we use here are normal Rust data types.
+CXX-Qt will automatically convert these types to their C++/Qt equivalent.
+In our case that means:
+- `number: i32` -> `int number`
+- `string: String` -> `QString string`\
+For more details on the available types, see the [Qt types page](../concepts/types.md).
```

Comment:
the formatting looks odd here. Think you need a blank line at the end of the list ?

Also wonder if there is any difference between these two (note i normally indent the bullet point as well.
```
- thing
- thing
```
vs
```
  * thing
  * thing
```

Reply:
Yup, the trailing backslash is intentional here.
It makes the "For more details" line indented on the same level as the enumeration.

I think it looks okay, because it saves us from adding another entire paragraph.

Was considering using footnotes as well, but thought this was an okay compromise.

Reply:
OK lets keep it like this.